libdnf/goal/Goal.cpp

```diff
@@ -1167,6 +1167,49 @@ Goal::listObsoletedByPackage(DnfPackage *pkg)
     return pset;
 }
 
+static std::string string_join(const std::vector<std::string> & src, const std::string & delim)
+{
+    if (src.empty()) {
+        return {};
+    }
+    std::string output(*src.begin());
+    for (auto iter = std::next(src.begin()); iter != src.end(); ++iter) {
+        output.append(delim);
+        output.append(*iter);
+    }
+    return output;
+}
+
+std::string
+Goal::formatAllProblemRules(const std::vector<std::vector<std::string>> & problems)
+{
+    if (problems.empty()) {
+        return {};
+    }
+    bool single_problems = problems.size() == 1;
+    std::string output;
+
+    if (!single_problems) {
```

Comment:
Is the condition correct? Really if "not" single_problems?

Reply:
Fixed